earthpy/tests/conftest.py

```diff
@@ -0,0 +1,91 @@
+""" Utility functions for tests. """
+import numpy as np
+import pytest
+from shapely.geometry import Polygon, Point, LineString
+import geopandas as gpd
+
+
+def make_locs_gdf():
+    """ Create a dummy point GeoDataFrame. """
+    pts = np.array([[2, 2],
+                    [3, 4],
+                    [9, 8],
+                    [-12, -15]])
+    gdf = gpd.GeoDataFrame([Point(xy) for xy in pts],
+                           columns=['geometry'],
+                           crs={'init': 'epsg:4326'})
+    return gdf
+
+
+def make_poly_in_gdf():
+    """ Bounding box polygon. """
+    poly_inters = Polygon([(0, 0), (0, 10), (10, 10), (10, 0), (0, 0)])
+    gdf = gpd.GeoDataFrame([1],
+                           geometry=[poly_inters],
+                           crs={'init': 'epsg:4326'})
+    return gdf
+
+
+def make_locs_buff():
+    """ Buffer points to create multi poly. """
+    buffered_locations = make_locs_gdf()
+    buffered_locations["geometry"] = buffered_locations.buffer(4)
+    return buffered_locations
+
+
+def make_donut_geom():
+    """ Make a donut geometry. """
+    donut = gpd.overlay(make_locs_buff(), make_poly_in_gdf(),
+                        how='symmetric_difference')
+    return donut
+
+
+@pytest.fixture
```

Comment:
this is neat. what does this @pytest.fixture call actually do for this function? i'm more used to this in class definitions. and once this is setup how do we call these data in the tests?

Reply:
By adding `@pytest.fixture` as a decorator for a function in `conftest.py`, we are making the `locs_gdf` function available for all `earthpy/tests/test_*.py` modules. This is a way to generate data that are shared by multiple tests that span different test modules. See here for more: https://docs.pytest.org/en/latest/fixture.html#conftest-py-sharing-fixture-functions

Reply:
Examples of calling these fixtures in tests can be found in `test_clip.py`, where they show up as function arguments.

Reply:
perfect. ok i get it. thanks!!